recipes/easy_profiler/all/conanfile.py

```diff
@@ -25,72 +30,85 @@ class EasyProfilerConan(ConanFile):
+                    f"{self.ref} requires C++{self._min_cppstd}, which your compiler does not support."
+                )
+        if is_msvc_static_runtime(self) and self.options.shared and Version(self.settings.compiler.version) >= "15":
             raise ConanInvalidConfiguration(
-                "{} {} with MTd runtime not supported".format(self.settings.compiler,
-                                                              self.settings.compiler.version)
+                "{} {} with static runtime not supported".format(self.settings.compiler,
+                                                                 self.settings.compiler.version)
             )
 
     def source(self):
-        tools.get(**self.conan_data["sources"][self.version],
-                  destination=self._source_subfolder, strip_root=True)
+        get(self, **self.conan_data["sources"][self.version], strip_root=True)
 
+    def generate(self):
+        tc = CMakeToolchain(self)
+        tc.variables["EASY_PROFILER_NO_GUI"] = True
+        tc.variables["EASY_PROFILER_NO_SAMPLES"] = True
+        tc.generate()
+
+        deps = CMakeDeps(self)
+        deps.generate()
+        
     def build(self):
-        for patch in self.conan_data.get("patches", {}).get(self.version, []):
-            tools.patch(**patch)
-        cmake = self._configure_cmake()
+        apply_conandata_patches(self)
+        cmake = CMake(self)
+        cmake.configure()
         cmake.build()
 
-    def _configure_cmake(self):
-        if self._cmake:
-            return self._cmake
-        self._cmake = CMake(self)
-        # Don't build the GUI because it is dependent on Qt
-        self._cmake.definitions["EASY_PROFILER_NO_GUI"] = True
-        self._cmake.definitions["EASY_PROFILER_NO_SAMPLES"] = True
-        self._cmake.configure(build_folder=self._build_subfolder)
-        return self._cmake
-
     def package(self):
-        self.copy("LICENSE", dst="licenses", src=self._source_subfolder)
-        self.copy("LICENSE.MIT", dst="licenses", src=self._source_subfolder)
-        self.copy("LICENSE.APACHE", dst="licenses", src=self._source_subfolder)
-        cmake = self._configure_cmake()
+        copy(self, pattern="LICENSE", dst=os.path.join(self.package_folder,"licenses"), src=self.source_folder)
+        copy(self, pattern="LICENSE.MIT", dst=os.path.join(self.package_folder,"licenses"), src=self.source_folder)
+        copy(self, pattern="LICENSE.APACHE", dst=os.path.join(self.package_folder,"licenses"), src=self.source_folder)
+        cmake = CMake(self)
         cmake.install()
-        tools.rmdir(os.path.join(self.package_folder, "lib", "cmake"))
-        os.remove(os.path.join(self.package_folder, "LICENSE.MIT"))
-        os.remove(os.path.join(self.package_folder, "LICENSE.APACHE"))
+
+        rmdir(self, os.path.join(self.package_folder, "lib", "cmake"))
+        rm(self, "LICENSE.MIT", self.package_folder)
+        rm(self, "LICENSE.APACHE", self.package_folder)
```

Comment:
me nit picking but we copy above and delete them here? maybe cut two lines above and make this a move/rename?

less is more ❤️

Reply:
Seems these línies are necessary to avoid a hoock error```

```

[HOOK - conan-center.py] post_package(): ERROR: [DEFAULT PACKAGE LAYOUT (KB-H013)] Unknown file 'LICENSE.MIT' in the package (https://github.com/conan-io/conan-center-index/blob/master/docs/error_knowledge_base.md#KB-H013-DEFAULT-PACKAGE-LAYOUT) 
[HOOK - conan-center.py] post_package(): ERROR: [DEFAULT PACKAGE LAYOUT (KB-H013)] Unknown file 'LICENSE.APACHE' in the package (https://github.com/conan-io/conan-center-index/blob/master/docs/error_knowledge_base.md#KB-H013-DEFAULT-PACKAGE-LAYOUT) 
```

I guess that otherwise the license files get places in the root of the package.